meshtastic/node_runtime/content_runtime.py

```
@@ -0,0 +1,496 @@
+    def _has_routing_error(decoded: dict[str, Any]) -> bool:
+        """Return True when decoded routing payload contains a non-NONE error reason."""
+        routing = decoded.get("routing")
+        if not isinstance(routing, dict):
+            return False
+        error_reason = routing.get("errorReason")
+        if isinstance(error_reason, str) and error_reason != "NONE":
+            logger.error("Error on response: %s", error_reason)
+            return True
+        return False
+
+    def handle_ringtone_response(
```

Ruff (0.15.6)

[warning] Too many return statements (7 > 6)

(PLR0911)